server/schema/user/user.datasources.js

```diff
@@ -94,6 +94,17 @@ const search = (query, accountType, limit, offset) =>
     },
   ]);
 
+const getIdByOldUserName = async (oldUserName) => {
+  try {
+    const _users = await UserModel.find({ oldUserName: oldUserName });
+    const [{ _id: id }] = _users;
```

Comment:
Just get the full user in datasources and prime it by ID. No need to return just the ID.
See article datasources for reference.